src/app/planner/containers/planner.component.ts

```diff
@@ -56,4 +73,51 @@ export class PlannerComponent implements OnInit {
                 });
             }).unsubscribe();
     }
+
+    public eventModify(event) {
+        if (!this.confirmationDialogOpen) {
+            this.confirmationDialogRef = this.openConfirmationDialog();
+            this.confirmationDialogRef.afterClosed().subscribe(confirmed => {
+                if (confirmed) {
+                    this.sessionCreationService.update(this.buildSessionUpdate(event));
+
+                    this.openSummaryDialog();
+                } else {
+                    event.detail.revertFunc();
+                }
+                this.confirmationDialogOpen = false;
+            });
+        }
+    }
+
+    private buildSessionUpdate(event) {
+        return {
+            id: event.detail.event.id,
+            start: event.detail.event.start.toDate(),
+            duration: moment.duration(event.detail.event.end.diff(event.detail.event.start)).asSeconds(),
```

Comment:
as I wrote above, now we are assuming there will be fullcalendar event with our session details but we are emitting 'any' type anyway, a thing to think about - are we secure, will we be able to test and maintain this etc.

Reply:
The problem is you can modify all properties of a session, one of them, or a few of them. We'd have to define an interface like 'SessionUpdate' that's entirely the same as 'Session' but with all fields marked as optional. Or we can simply assume the type being 'Session' here and set only the fields that should be modified. I'll think about it in the next story.